domain_config/src/DomainConfigOverrider.php

```diff
@@ -165,18 +172,20 @@ public function getCacheableMetadata($name) {
   protected function initiateContext() {
     // Prevent infinite lookups by caching the request. Since the _construct()
     // is called for each lookup, this is more efficient.
-    static $context;
-    if ($context) {
-      return;
-    }
-    $context++;
+    $this->contextSet = TRUE;
```

Comment:
The patch splits this check into 2 methods. I think it's better to keep everything here as getCacheableMetadata() also calls initiateContext.

If a check outside of the method is required a change to a class static and self::$context could be a better way to go.

Reply:
I can't recall what in the call stack made it possible that Cache would be initiated without running the overrides. Possibly registration of the event handler.

Are there use-cases for calling this outside of it's current scope? Perhaps for modules trying to deal with the hook issue?